app/src/androidTest/java/com/github/sdpcoachme/CachingDatabaseTest.kt

```diff
@@ -33,44 +38,144 @@ class CachingDatabaseTest {
     fun addUserPutsUserInCache() {
         val wrappedDatabase = MockDatabase()
         val cachingDatabase = CachingDatabase(wrappedDatabase)
-        val email = "oui@non.com"
-        val user = UserInfo(
-            "Will",
-            "Smith",
-            email,
-            "0000000000",
-            NEW_YORK,
-            false,
-            emptyList(),
-            emptyList()
-        )
-        val retrievedUser = cachingDatabase.updateUser(user)
-            .thenCompose { cachingDatabase.getUser(email) }
+        val retrievedUser = cachingDatabase.updateUser(willSmithUser)
+            .thenCompose { cachingDatabase.getUser(willSmithUser.email) }
             .get(5, SECONDS)
-        assertTrue(cachingDatabase.isCached(email))
-        assertEquals(user, retrievedUser)
+        assertTrue(cachingDatabase.isCached(willSmithUser.email))
+        assertTrue(cachingDatabase.userExists(willSmithUser.email).get(1, SECONDS))
+        assertEquals(willSmithUser, retrievedUser)
     }
 
     @Test
     fun addUserOverridesPreviousValueInCache() {
         val wrappedDatabase = MockDatabase()
         val cachingDatabase = CachingDatabase(wrappedDatabase)
-        val newUser = UserInfo(
-            "John",
-            "Doe",
-            exampleEmail,
-            "1234567890",
-            LAUSANNE,
-            false,
-            emptyList(),
-            emptyList()
-        )
         val updatedUser = cachingDatabase.getUser(exampleEmail)
             .thenCompose {
-                cachingDatabase.updateUser(newUser) }
+                cachingDatabase.updateUser(defaultUser) }
             .thenCompose { cachingDatabase.getUser(exampleEmail) }
             .get(5, SECONDS)
         assertTrue(cachingDatabase.isCached(exampleEmail))
-        assertEquals(newUser, updatedUser)
+        assertEquals(defaultUser, updatedUser)
+    }
+
+    @Test
+    fun addAllUsersPutsAllUsersInCache() {
```

Comment:
I am not sure if I understood what you want to test here. I think it makes sense to test those things, but rather in two separated tests. 
We can have one test where you create the wrappedDatabase, then add all the users to it, then create the cachingDatabase, and then call getAllUsers and check if the users are cached. This test will test the caching functionality of getAllUsers.
As a second test, you can check if calling getAllUsers caches the users correctly.

Reply:
Actually here the goal is to test if the getAllUsers method puts all the users in the cache. For that, we first have to put all users in the database and clear the cache.
Another test checks if the method updateUser puts the user in the cache.

Therefore, I changed this test's name to getAllUsersPutsAllUsersInCache